contentpool_client.module

```diff
@@ -1,6 +1,82 @@
 <?php
 
+use Drupal\Core\Form\FormStateInterface;
+
 /**
  * @file
  * Module file.
  */
+
+/**
+ * Implements HOOK_form_FORM_ID_alter.
+ */
+function contentpool_client_form_remote_edit_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $remote = $form_state->getFormObject()->getEntity();
+
+  $form['actions']['submit']['#submit'] = array_merge(['contentpool_client_form_remote_edit_form_submit'], $form['actions']['submit']['#submit']);
+
+  $form['contentpool_client_settings'] = [
+    '#type' => 'details',
+    '#title' => t('Contentpool client settings'),
+    '#open' => TRUE,
+  ];
+
+  $form['contentpool_client_settings']['autoregister'] = [
+    '#type' => 'checkbox',
+    '#title' => t('Autoregister this site to remote site (if supported)'),
+    '#description' => t('If the remote site is a contentpool server, the site can be autoregistered remotely.'),
```

Comment:
We always want the autoregister - so that the contentpool always has a complete list of clients. so let's remove this option.

Reply:
I don't think we should expect every remote to be a contentpool server. Also if a contentpool server is down this could be useful to avoid the relentless checking for connection.

I think we should keep the option but maybe also add some kind of keep-alive logic that marks clients which loose their connection, go down, or deactivate their autoregistering as inactive on the contentpool-server. What do you think?

Reply:
>I think we should keep the option but maybe also add some kind of keep-alive logic that marks clients which loose their connection, go down, or deactivate their autoregistering as inactive on the contentpool-server. What do you think?

That makes a ton of sense. I see this going to happen with test sites which we could connect to a central test-contentpool. This will lead to a lots of entries, what's ok - but properly handling inactive ones would be important then!

>I don't think we should expect every remote to be a contentpool server. Also if a contentpool server is down this could be useful to avoid the relentless checking for connection.

Good point, I did not think of this. I'd not care much about the contentpool down scenario, I think it's safe to except that it's up and to display errors else. 

So I guess what this should be the then is a flag to tell us this is a contentpool server. If so, then we assume the autoregister logic. But we might need this contentpool flag information in some other scenarios in future, so let's better label it with some like "Is contentpool" and "Whether the remote site is an instance of the contentpool distribution."